src/parser.js

```diff
@@ -658,6 +654,20 @@ export default class Parser extends Stream {
     });
   }
 
+  warnOnMissingAttributes_(identifier, attributes, required) {
```

Comment:
Helper to warn on missing attributes, found myself repeating the same code in a ton of places.